back-nest/src/getAllElement/products.services.ts

```diff
@@ -0,0 +1,39 @@
+import { BadRequestException, Injectable } from '@nestjs/common';
+import { InjectRepository } from '@nestjs/typeorm';
+import { Repository } from 'typeorm';
+import { Product } from '../models/product.enity.model';
+import { RedisService } from '../redis/redis.service';
+
+function sleep(ms) {
+  return new Promise((resolve) => setTimeout(resolve, ms));
+}
+
+@Injectable()
+export class ProductService {
+  constructor(
+    @InjectRepository(Product)
+    private readonly productRepository: Repository<Product>,
+    private readonly redisService: RedisService,
+  ) {}
+
+  async getAllProducts(): Promise<Product[]> {
+    const cachedProducts =
+      await this.redisService.get<Product[]>('all_products');
```

Comment:
I suggest to move `all_products` cache key into a constant file, so that you can use it from one place and don't mess up in future. Imagine if you have more than 100 cache keys in your project. Managing them without constant would be a pain in ass.

Here's an example
```ts
export const cacheKeys = {
  PRODUCTS: 'all_products' // use for products caching
}
```

```suggestion
      await this.redisService.get<Product[]>(cacheKeys.PRODUCTS);
```